plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -159,10 +162,27 @@ public Tuple<BlobServiceClient, Supplier<Context>> client(String clientName, BiC
         return new Tuple<>(state.getClient(), () -> buildOperationContext(azureStorageSettings));
     }
 
+    public StorageEndpoint getStorageBlobEndpoint(final AzureStorageSettings settings) {
```

Comment:
Any reason why `getStorageBlobEndpoint` should be public? 
I see tests are using it, can we expose it via annotations [1]

[1] https://javadoc.io/doc/org.jetbrains/annotations/22.0.0/org/jetbrains/annotations/VisibleForTesting.html

Reply:
Thank you @saratvemulapalli, there is no specific reason why I make it a public, maybe I am just thinking that it is a standard blob storage endpoint, we are only exposing the account name, and nothing else. It can be private.

I did not manage to find usage of `org.jetbrains.annotations.VisibleForTesting` in the opensearch code base, so i have changed it to `private` instead. And i have amended my tests to accommodate this change without loosing test coverage.